src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -452,7 +452,10 @@ def extract_query_filters(self, cmd):
         return FilterSpec(mds_ranks, client_id, client_ip)
 
     def get_perf_data(self, cmd):
-        filter_spec = self.extract_query_filters(cmd)
+        try:
+            filter_spec = self.extract_query_filters(cmd)
+        except ValueError:
+            return -errno.EINVAL, "", "invalid filter"
```

Comment:
'invalid mds rank' sounded more user friendly, right ?

Reply:
We are considering ValueError from all filters.
If ValueError occurs for any other filter in  extract_query_filters then 'invalid mds rank' will be misleading.
Hence considering ValueError for all filters.

Reply:
Yes,  but user wouldn't understand what a filter is. The user passes mds_rank, client_id and client_ip. As of now, only mds_rank does validation and not client_id/client_ip. If you want to make it future proof, what about something like 'invalid mds_rank or client_ip'. This is more obvious for the user.

Reply:
Changed message to 'invalid mds_rank'